server.go

```diff
@@ -603,6 +604,7 @@ func (r *oauthProxy) createUpstreamProxy(upstream *url.URL) error {
 
 	// create the forwarding proxy
 	proxy := goproxy.NewProxyHttpServer()
+	proxy.KeepDestinationHeaders = true
```

Comment:
Why not, but what is the use case for value false? In the current setup, headers setup by the middleware are just thrown away in the relayed response from upstream (cors or not cors btw).  I am not convinced of any proper usage with a "false" setting. Further, setting up CORS, and defaulting to making it unusable for upstream looks at best difficult to explain to a regular user.

Reply:
I am changing this to making it settable as you suggested, but I'd rather change the default, which would otherwise be misleading IHMO.
Like this: 
```golang
proxy.KeepDestinationHeaders = !cfg.CorsDisableUpstream
```